src/TraceEvent/TraceLog.cs

```diff
@@ -3671,6 +3672,7 @@ private unsafe void InitializeFromFile(string etlxFilePath)
         }
 
         private static char[] s_directorySeparators = { '\\', '/' };
+        private static string[] s_validExtensions = { ".dll", ".exe" };
```

Comment:
This could be `HashSet<string>` for faster lookups (not that it is in super hot path, but still). And also use some explicit _string comparer_, probably case insensitive makes sense here.